cmd/zc_filter.go

```diff
@@ -230,6 +231,107 @@ func (fs filterSet) GetEnumerationPreFilter(recursive bool) string {
 
 ////////
 
+// includeRegex includes files that follow the regex expression
+type includeRegex struct {
+	patterns []string
+}
+
+func (f *includeRegex) doesSupportThisOS() (msg string, supported bool) {
+	msg = ""
+	supported = true
+	return
+}
+
+func (f *includeRegex) appliesOnlyToFiles() bool {
+	return true
+}
+
+func (f *includeRegex) doesPass(storedObject storedObject) bool {
+	if len(f.patterns) == 0 {
+		return true
+	}
+	for _, pattern := range f.patterns {
+		matched := false
+		var err error
+
+		matched, err = regexp.MatchString(pattern, storedObject.name)
+
+		if err != nil {
+			return false
+		}
+		if matched {
+			return true
+		}
+	}
+	return false
+}
+
+func buildIncludeRegexFilters(patterns []string) []objectFilter {
+	if len(patterns) == 0 {
+		return []objectFilter{}
+	}
+
+	validPatterns := make([]string, 0)
+	for _, pattern := range patterns {
+		if pattern != "" {
+			validPatterns = append(validPatterns, pattern)
+		}
+	}
+
+	return []objectFilter{&includeRegex{patterns: validPatterns}}
+}
+
+// excludeRegex does not include files that match with the regex expression
+type excludeRegex struct {
```

Comment:
You do not need two types for this. Just add a flag in includeRegex to set it up as an exclude filter. This is otherwise duplicated code.

Reply:
Perhaps even have it be like,

```go
type includeRegex struct {
    patterns []string
    matchResponse bool // true = include, false = exclude
}
```

and return `matchResponse` when there's a regex match, but `!matchResponse` when there's a hit